main/lsp/requests/workspace_symbols.cc

```diff
@@ -86,15 +86,13 @@ vector<unique_ptr<SymbolInformation>> SymbolMatcher::symbolRef2SymbolInformation
         if (location == nullptr) {
             continue;
         }
+        // VSCode does its own internal ranking based on comparing the query string against the result name.
+        // Therefore have the name be the fully qualified name if it makes sense (e.g. Foo::Bar instead of Bar)
         auto result =
-            make_unique<SymbolInformation>(sym->name.show(gs), symbolRef2SymbolKind(gs, symRef), std::move(location));
-        if (!hideSymbol(gs, sym->owner)) {
-            // VSCode does its own internal ranking based on comparing the query string against the result name.
-            // Therefore have the name be the full path if it makes sense (e.g. Foo::Bar instead of Bar) so fully
-            // qualified symbol search works. Remove leading :: for readability
-            auto fully_qualified_name = sym->showFullName(gs);
-            result->name = fully_qualified_name.erase(0, fully_qualified_name.find_first_not_of(":"));
-            result->containerName = sym->owner.data(gs)->showFullName(gs);
+            make_unique<SymbolInformation>(sym->show(gs), symbolRef2SymbolKind(gs, symRef), std::move(location));
+        auto container = sym->owner;
+        if (container != core::Symbols::root()) {
```

Comment:
Shouldn't this be `!hideSymbol(gs, sym->owner)`? I think that covers a few extra cases besides root owner.

Reply:
It does, but those are either inapplicable (Stubs shouldn't show up as a search result anyway) or actively wrong in this case (for example, Class methods return true for hidesymbol)